balancer-js/src/modules/pools/join/join.module.ts

```diff
@@ -0,0 +1,39 @@
+import { Pools } from '@/modules/pools/pools.module';
+import { PoolType } from '@/types';
+import { JoinPoolAttributes } from '../pool-types/concerns/types';
+
+export class Join {
+  private pools: Pools;
+
+  constructor(pools: Pools) {
+    this.pools = pools;
+  }
+
+  /**
+   * Build join pool transaction parameters with exact tokens in and minimum BPT out based on slippage tolerance
+   * @param {string}    joiner - Address used to join pool
+   * @param {string}    poolId - Id of pool being joined
+   * @param {string[]}  tokensIn - Token addresses provided for joining pool (same length and order as amountsIn)
+   * @param {string[]}  amountsIn - Token amounts provided for joining pool in EVM amounts
+   * @param {string}    slippage - Maximum slippage tolerance in bps i.e. 50 = 0.5%
+   * @returns           transaction request ready to send with signer.sendTransaction
+   */
+  async buildExactTokensInJoinPool(
```

Comment:
Should we rename to just `join`? Similar to example - idea is to keep it simple for user.

Reply:
And how will we differ from different types of joins?
By types of joins I mean:
- exact tokens in for BPT out
- tokens in for exact BPT out
- ...
Should we add an input param with the join type?

Reply:
At the moment the front-end only uses exact tokens in. Unless we get some requests for the other type I think we can leave it to focus on other things for now.

Reply:
Got it - I'll refactor to just `join` over here then.

Reply:
done ✅